Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -1038,6 +1073,42 @@ public struct Timestamp: BSONValue, Equatable, Codable {
     }
 }
 
+/// A struct to represent the deprecated Undefined type.
+/// Undefined instances cannot be created, but they can be read from existing documents that contain them.
+public struct BSONUndefined: BSONValue, Equatable, Codable {
```

Comment:
re: `@deprecated`, see my other comment on `DBPointer`.

I'd rather just do it all in one swoop rather than having this one type be separate, if that's okay.